### PR TITLE
Documentation improvements to sampling operators

### PR DESCRIPTION
**Pull request by madjam**

### CODE REVIEW

File: src/operator/tensor/sample_op.cc [hunk 1]
```diff
@@ -31,52 +31,42 @@ MXNET_OPERATOR_REGISTER_SAMPLE(random_uniform, SampleUniformParam)
 .add_alias("uniform")
 .add_alias("_sample_uniform")
 .describe(R"code(Draw samples from a uniform distribution.
-
 Samples are uniformly distributed over the half-open interval [low, high)
-(includes low, but excludes high)::
-
-  nd.uniform(low=0, high=1, shape=(2,2)) = [[ 0.60276335,  0.85794562],
```
Comment (piiswrong) [inline src/operator/tensor/sample_op.cc]: why remove the example?

Comment (madjam) [inline src/operator/tensor/sample_op.cc]: - I thought it was simple enough that we didn't need an example for clarification.
- Also the code sample generates random samples, it is not reproducible by users and by unit tests
- Other sampling operators do not have examples.

If you feel strongly about having an example, we can add it back.

Comment (madjam) [inline src/operator/tensor/sample_op.cc]: @piiswrong are you ok with this in this state?

File: src/operator/tensor/sample_op.h [hunk 2]
```diff
@@ -25,24 +25,27 @@ struct SampleUniformParam : public dmlc::Parameter<SampleUniformParam> {
...
               "Only used for imperative calls.");
     DMLC_DECLARE_FIELD(dtype)
     .add_enum("None", -1)
     .add_enum("float32", mshadow::kFloat32)
     .add_enum("float64", mshadow::kFloat64)
     .add_enum("float16", mshadow::kFloat16)
     .set_default(-1)
-    .describe("DType of the output. If output given, set to type of output."
-              "If output not given and type not defined (dtype=None), set to float32.");
+    .describe("DType of the output. If output is specified, set to `dtype` of output. "
```
Comment (zackchase) [inline src/operator/tensor/sample_op.h]: should DType be capitalized in this fashion?

File: src/operator/tensor/sample_op.h [hunk 3]
```diff
@@ -54,24 +57,25 @@ struct SampleNormalParam : public dmlc::Parameter<SampleNormalParam> {
...
```
Comment (zackchase) [inline src/operator/tensor/sample_op.h]: see above

File: src/operator/tensor/sample_op.h [hunk 4]
```diff
@@ -83,24 +87,25 @@ struct SampleGammaParam : public dmlc::Parameter<SampleGammaParam> {
...
```
Comment (zackchase) [inline src/operator/tensor/sample_op.h]: ""

File: src/operator/tensor/sample_op.h [hunk 5]
```diff
@@ -111,22 +116,23 @@ struct SampleExponentialParam : public dmlc::Parameter<SampleExponentialParam> {
...
```
Comment (zackchase) [inline src/operator/tensor/sample_op.h]: ""

File: src/operator/tensor/sample_op.h [hunk 6]
```diff
@@ -164,24 +171,25 @@ struct SampleNegBinomialParam : public dmlc::Parameter<SampleNegBinomialParam> {
...
```
Comment (zackchase) [inline src/operator/tensor/sample_op.h]: ""

**Comment by madjam** [comment]
Withdrawing in favor of #6094

**Pull request closed without merging**